pepdbagent/modules/project.py

```diff
@@ -290,28 +330,40 @@ def _overwrite(
         namespace = namespace.lower()
         if self.exists(namespace=namespace, name=proj_name, tag=tag):
             _LOGGER.info(f"Updating {proj_name} project...")
-            with self._sa_engine.begin() as conn:
-                conn.execute(
-                    update(Projects)
-                    .values(
-                        namespace=namespace,
-                        name=proj_name,
-                        tag=tag,
-                        digest=project_digest,
-                        project_value=project_dict,
-                        number_of_samples=number_of_samples,
-                        private=private,
-                        last_update_date=datetime.datetime.now(datetime.timezone.utc),
-                        pep_schema=pep_schema,
+            statement = self._create_select_statement(proj_name, namespace, tag)
+            with Session(self._sa_engine) as session:
+                found_prj = session.scalars(statement).one()
+
+                if found_prj:
+                    _LOGGER.debug(
+                        f"Project has been found: {found_prj.namespace}, {found_prj.name}"
                     )
-                    .where(
-                        and_(
-                            Projects.namespace == namespace,
-                            Projects.name == proj_name,
-                            Projects.tag == tag,
-                        )
+
+                    found_prj.digest = project_digest
+                    found_prj.number_of_samples = number_of_samples
+                    found_prj.private = private
+                    found_prj.pep_schema = pep_schema
+
+                    # Deleting old samples and subsamples
+                    if found_prj.samples_mapping:
+                        for sample in found_prj.samples_mapping:
+                            _LOGGER.debug(f"deleting samples: {str(sample)}")
+                            session.delete(sample)
```

Comment:
Is there a way to delete more than 1 at time instead of looping like this?

eg, `DELETE WHERE ... `

Reply:
This is the easiest way how to do it using sqlalchemy that I found. Not sure if it's possible using declarative object